src/main/java/uk/gov/hmcts/ccd/domain/service/common/AccessControlService.java

```diff
@@ -239,6 +252,46 @@ public CaseEventTrigger setReadOnlyOnCaseViewFieldsIfNoAccess(final CaseEventTri
         return caseEventTrigger;
     }
 
+    public CaseEventTrigger updateCollectionDisplayContextParameterByAccess(final CaseEventTrigger caseEventTrigger,
+                                                                            final Set<String> userRoles) {
+        caseEventTrigger.getCaseFields()
+            .forEach(caseViewField -> {
+                if (caseViewField.isCollectionFieldType()) {
+                    caseViewField.setDisplayContextParameter(getCollectionAccess(userRoles, caseViewField));
+                }
+
+                setChildrenCollectionDisplayContextParameter(caseViewField.getFieldType().getChildren(), userRoles);
+            });
+        return caseEventTrigger;
+    }
+
+    private void setChildrenCollectionDisplayContextParameter(final List<CaseField> caseFields,
+                                                              final Set<String> userRoles) {
+        caseFields.forEach(childField -> {
+            if (childField.isCollectionFieldType()) {
+                childField.setDisplayContextParameter(getCollectionAccess(userRoles, childField));
+            }
+
+            setChildrenCollectionDisplayContextParameter(childField.getFieldType().getChildren(), userRoles);
+        });
+    }
+
+    private String getCollectionAccess(Set<String> userRoles, CommonField field) {
+        List<String> collectionAccess = new ArrayList<>();
+        if (hasAccessControlList(userRoles, CAN_CREATE, field.getAccessControlLists())) {
```

Comment:
we need to be careful here. Strictly speaking this ticket should only have effect in case of case roles. But because userRoles also contains idam roles, when EUI implements the UI part of this ticket, it's possible the UI behaviour will change also for permissions set on Idam roles...

Reply:
what I mean is this code adds COLLECTION(allowInsert) if there is an idam role with C permission, even if there's no case roles. We need to discuss about this being correct

Reply:
As discussed I will put a comment for ExUI this is for both Case and Idam roles